File: frontend/src/main/ng/src/app/service/metric.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Profile } from '../model/profile';
import { Metric } from '../model/metric';
import { environment } from '../../environments/environment';
import { Recommendation } from '../model/recommendation';

@Injectable({
  providedIn: 'root'
})
export class MetricService {

  private newsUrl = `${environment.API_URL}/api/v1/metric`;

  constructor(private http: HttpClient) { }

  getProfile(symbol: string) {
    const url = `${this.newsUrl}/profile/${symbol}`;
    return this.http.get<Profile>(url);
  };

  getMetrics(symbol: string) {
    const url = `${this.newsUrl}/metric/${symbol}`;
    return this.http.get<Metric>(url);
  };

  getRecommendations(symbol: string) {
    const url = `${this.newsUrl}/recommendation/${symbol}`;
    return this.http.get<Recommendation[]>(url);
  }
}

File: backend/src/main/java/eye/on/the/money/service/MetricService.java
package eye.on.the.money.service;

import eye.on.the.money.model.stock.Metric;
import eye.on.the.money.model.stock.Profile;
import eye.on.the.money.model.stock.Recommendation;

import java.util.List;

public interface MetricService {
    public Profile getProfileBySymbol(String symbol);
    public Metric getMetricBySymbol(String symbol);
    public List<Recommendation> getRecommendations(String symbol);
}
